src/loaders/OmeZarrLoader.ts

```diff
@@ -415,8 +508,12 @@ class OMEZarrLoader extends ThreadableVolumeLoader {
     const channelPromises = channelIndexes.map(async (ch) => {
       // Build slice spec
       const { min, max } = regionPx;
-      const unorderedSpec = [loadSpec.time, ch, slice(min.z, max.z), slice(min.y, max.y), slice(min.x, max.x)];
-      const sliceSpec = this.orderByDimension(unorderedSpec as TCZYX<number | Slice>);
+      const { sourceIndex: sourceIdx, channelIndexInSource: sourceCh } = this.matchChannelToSource(ch);
+      const unorderedSpec = [loadSpec.time, sourceCh, slice(min.z, max.z), slice(min.y, max.y), slice(min.x, max.x)];
+
+      const level = this.sources[sourceIdx].scaleLevels[levelIdx];
+      const sliceSpec = this.orderByDimension(unorderedSpec as TCZYX<number | Slice>, sourceIdx);
+      const reportKey = (key: string, sub: SubscriberId) => reportKeyBase(sourceIdx, key, sub);
```

Comment:
this is a pretty generic comment, but looking at this on GitHub it seems like this is still part of `loadRawChannelData`? In that case, this is a very long function that just got longer. could this "build slice spec" be it's own function ?

Reply:
I took a second look at `loadRawChannelData`, and the easiest part to factor out seemed to be the `ImageInfo`-updating code at the beginning. So I moved that section to its own function. The remaining code in `loadRawChannelData` is still pretty high cognitive load (lots of nested scopes, plus this `async` closure that gets called a bunch of times) but it's all tougher to separate. Might come back later and try though.